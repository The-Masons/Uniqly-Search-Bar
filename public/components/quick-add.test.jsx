import React from 'react';
import QuickAdd from './quick-add.jsx';

describe('QuickAdd', () => {
  test('should populate dropdowns with given data', () => {
    const quickAdd = mount(
      <QuickAdd sizes={[
        'Fake Size 0',
        'Fake Size 1',
        'Fake Size 2',
      ]}
      quantities={{
        'Fake Size 0': 10,
        'Fake Size 1': 11,
        'Fake Size 2': 12,
      }}/>
    );

    expect(quickAdd.containsAllMatchingElements([
      <option key='sizeOpt0'>Fake Size 0</option>,
      <option key='sizeOpt1'>Fake Size 1</option>,
      <option key='sizeOpt2'>Fake Size 2</option>,
    ])).toEqual(true);
  });

  test('should populate quantity dropdown with the correct number of options', () => {
    const quickAdd = mount(
      <QuickAdd sizes={[
        'Fake Size 0',
        'Fake Size 1',
        'Fake Size 2',
      ]}
      quantities={{
        'Fake Size 0': 10,
        'Fake Size 1': 11,
        'Fake Size 2': 12,
      }}/>
    );

    expect(quickAdd.find('.quickadd-select-quantity').children().length).toEqual(10);
    quickAdd.find('.quickadd-select-sizes').simulate('change', {target: {form: [{value: 'Fake Size 2'}]}});
    expect(quickAdd.state('currentSize')).toEqual('Fake Size 2');
    expect(quickAdd.find('.quickadd-select-quantity').children().length).toEqual(12);
  });

  test('should add an item to the cart when "ADD TO BAG" is clicked', () => {
    const mockAdd = jest.fn();
    const quickAdd = mount(
      <QuickAdd sizes={[
        'Fake Size 0',
        'Fake Size 1',
        'Fake Size 2',
      ]}
      quantities={{
        'Fake Size 0': 10,
        'Fake Size 1': 11,
        'Fake Size 2': 12,
      }}
      addToCart={mockAdd}/>
    );

    quickAdd.find('.quickadd-btn').simulate('click', {target: {form: [{value: 'Fake Size 0'}, {value: 1}]}});

    expect(mockAdd).toHaveBeenCalledWith('Fake Size 0', 1);
  });
});
