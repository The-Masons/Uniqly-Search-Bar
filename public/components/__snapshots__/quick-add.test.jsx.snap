// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuickAdd should populate dropdowns with given data 1`] = `
<QuickAdd
  quantities={
    Object {
      "Fake Size 0": 10,
      "Fake Size 1": 11,
      "Fake Size 2": 12,
    }
  }
  sizes={
    Array [
      "Fake Size 0",
      "Fake Size 1",
      "Fake Size 2",
    ]
  }
>
  <div
    className="quickadd"
  >
    <form
      className="quickadd-form"
    >
      <div
        className="quickadd-select"
      >
        <select
          className="quickadd-dropdown quickadd-select-sizes"
          onChange={[Function]}
          value="Fake Size 0"
        >
          <option
            key="sizeOpt0"
          >
            Fake Size 0
          </option>
          <option
            key="sizeOpt1"
          >
            Fake Size 1
          </option>
          <option
            key="sizeOpt2"
          >
            Fake Size 2
          </option>
        </select>
        <select
          className="quickadd-dropdown quickadd-select-quantity"
          onChange={[Function]}
          value={1}
        >
          <option
            key="qtyOpt0"
          >
            1
          </option>
          <option
            key="qtyOpt1"
          >
            2
          </option>
          <option
            key="qtyOpt2"
          >
            3
          </option>
          <option
            key="qtyOpt3"
          >
            4
          </option>
          <option
            key="qtyOpt4"
          >
            5
          </option>
          <option
            key="qtyOpt5"
          >
            6
          </option>
          <option
            key="qtyOpt6"
          >
            7
          </option>
          <option
            key="qtyOpt7"
          >
            8
          </option>
          <option
            key="qtyOpt8"
          >
            9
          </option>
          <option
            key="qtyOpt9"
          >
            10
          </option>
        </select>
      </div>
      <button
        className="quickadd-btn"
        onClick={[Function]}
      >
        ADD TO BAG
      </button>
    </form>
  </div>
</QuickAdd>
`;
